packages/nuxt/src/core/templates.ts

```diff
@@ -142,11 +158,14 @@ const adHocModules = ['router', 'pages', 'imports', 'meta', 'components', 'nuxt-
 export const schemaTemplate: NuxtTemplate = {
   filename: 'types/schema.d.ts',
   getContents: async ({ nuxt }) => {
-    const moduleInfo = nuxt.options._installedModules.map(m => ({
-      ...m.meta,
-      importName: m.entryPath || m.meta?.name
-    })).filter(m => m.configKey && m.name && !adHocModules.includes(m.name))
-
+    const moduleInfo = []
+    for (const m of nuxt.options._installedModules) {
+      const meta = {...m.meta}
+      const impName = m.entryPath || meta?.name
+      if (meta.configKey && meta.name && !adHocModules.includes(meta.name)) {
+         moduleInfo.push({...meta, importName: impName})
+      }
+    }
     const relativeRoot = relative(resolve(nuxt.options.buildDir, 'types'), nuxt.options.rootDir)
     const getImportName = (name: string) => (name[0] === '.' ? './' + join(relativeRoot, name) : name).replace(/\.\w+$/, '')
     const modules = moduleInfo.map(meta => [genString(meta.configKey), getImportName(meta.importName)])
```

Comment:
TODO: revisit this, as the type checks fail but the regular tests don't, which might indicate a false negative.

Reply:
Update: fixed, missed a line in the previous refactors